include/faabric/scheduler/ExecGraph.h

```diff
@@ -24,6 +24,10 @@ std::set<std::string> getExecGraphHosts(const ExecGraph& graph);
 
 std::vector<std::string> getMpiRankHostsFromExecGraph(const ExecGraph& graph);
 
+void getMigratedMpiRankHostsFromExecGraph(const ExecGraph& graph,
+                                          std::vector<std::string>& hostsBefore,
+                                          std::vector<std::string>& hostsAfter);
```

Comment:
Passing vectors to be filled is a slightly unintuitive interface for a `getXYZ` method. If you're populating a vector you could call it `populateXYZ` or `appendXYZ`, but instead this should really return the two lists. Returning a `std::pair` of vectors seems more intuitive.